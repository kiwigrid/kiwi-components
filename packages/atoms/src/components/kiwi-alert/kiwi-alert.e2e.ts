import { newE2EPage } from '@stencil/core/testing';

describe('kiwi-alert', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<kiwi-alert></kiwi-alert>');

    const element = await page.find('kiwi-alert');
    expect(element).toHaveClass('hydrated');
  });
});
