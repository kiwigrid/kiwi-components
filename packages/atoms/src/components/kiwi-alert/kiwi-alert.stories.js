import { html } from 'lit-html';
import notes from './readme.md';

export default {
  title: 'KiwiComponents/Alert',
  parameters: { docs: { description: { component: notes } } },
  decorators: [],
};

export const basic = (args) => {
  const { type, text } = args;

  return html` <kiwi-alert type=${type}>${text}</kiwi-alert>`;
};
basic.args = {
  text: 'This is an alert',
};
basic.argTypes = {
  type: {
    control: {
      type: 'select',
      options: ['info', 'warn', 'error', 'success'],
    },
    defaultValue: 'info',
  },
};
