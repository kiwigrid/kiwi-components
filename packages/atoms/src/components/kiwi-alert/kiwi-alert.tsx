import { Component, ComponentInterface, h, JSX, Prop } from '@stencil/core';

@Component({
  tag: 'kiwi-alert',
  styleUrl: 'kiwi-alert.css',
  shadow: false,
})
export class KiwiAlert implements ComponentInterface {
  /**
   * The type of the alert.
   */
  @Prop()
  type: 'info' | 'warn' | 'error' = 'info';

  render(): JSX.Element {
    let alertClass = '';
    switch (this.type) {
      case 'info':
        alertClass = 'alert-info';
        break;
      case 'warn':
        alertClass = 'alert-warning';
        break;
      case 'error':
        alertClass = 'alert-danger';
        break;
      default:
        const exhaustive: never = this.type;
        break;
    }
    return (
      <div class={`alert ${alertClass}`}>
        <slot />
      </div>
    );
  }
}
