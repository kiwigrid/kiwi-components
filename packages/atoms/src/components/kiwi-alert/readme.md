# kiwi-alert



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description            | Type                                       | Default  |
| -------- | --------- | ---------------------- | ------------------------------------------ | -------- |
| `type`   | `type`    | The type of the alert. | `"error" \| "info" \| "success" \| "warn"` | `'info'` |


## Dependencies

### Used by

 - [kiwi-toasts](../kiwi-toasts)

### Graph
```mermaid
graph TD;
  kiwi-toasts --> kiwi-alert
  style kiwi-alert fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
