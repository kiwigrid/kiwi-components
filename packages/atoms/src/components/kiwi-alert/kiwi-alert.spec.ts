import { newSpecPage } from '@stencil/core/testing';
import { KiwiAlert } from './kiwi-alert';

describe('kiwi-alert', () => {
  ['info', 'warn', 'error'].forEach((type) => {
    it(`renders ${type}`, async () => {
      const alert = await newSpecPage({
        components: [KiwiAlert],
        html: `
          <kiwi-alert type="${type}">
            <span>${type}</span>
          </kiwi-alert>
        `,
      });

      expect(alert.root).toMatchSnapshot(type);
    });
  });
});
