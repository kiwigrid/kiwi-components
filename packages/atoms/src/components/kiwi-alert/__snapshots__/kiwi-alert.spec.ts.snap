// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kiwi-alert renders error: error 1`] = `
<kiwi-alert class="alert alert-danger" type="error">
  <!---->
  <span>
    error
  </span>
</kiwi-alert>
`;

exports[`kiwi-alert renders info: info 1`] = `
<kiwi-alert class="alert alert-info" type="info">
  <!---->
  <span>
    info
  </span>
</kiwi-alert>
`;

exports[`kiwi-alert renders success: success 1`] = `
<kiwi-alert class="alert alert-success" type="success">
  <!---->
  <span>
    success
  </span>
</kiwi-alert>
`;

exports[`kiwi-alert renders warn: warn 1`] = `
<kiwi-alert class="alert alert-warning" type="warn">
  <!---->
  <span>
    warn
  </span>
</kiwi-alert>
`;
